src/vidpf.rs

```diff
@@ -385,21 +455,130 @@ impl From<VidpfServerId> for Choice {
 }
 
 /// Adjusts values of shares during the VIDPF evaluation.
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 struct VidpfCorrectionWord<W: VidpfValue> {
     seed: VidpfSeed,
     left_control_bit: Choice,
     right_control_bit: Choice,
     weight: W,
 }
 
+impl<'a, T, P, const SEED_SIZE: usize> ParameterizedDecode<&'a Mastic<T, P, SEED_SIZE>>
+    for VidpfCorrectionWord<VidpfWeight<T::Field>>
+where
+    T: Type,
+    P: Xof<SEED_SIZE>,
+{
+    fn decode_with_param(
+        decoding_parameter: &&'a Mastic<T, P, SEED_SIZE>,
```

Comment:
We could get rid of the double reference by instead implementing `ParameterizedDecode<Mastic<T, P, SEED_SIZE>>`.